### DISCUSSION TITLE
rewrites source supports array plz.

### DISCUSSION THREAD

**Discussion by takeItIzzy** [vercel/next.js#22401] [Ideas]
It will be useful when rewrite multiple source to one destination.
```javascript
module.exports = {
  async rewrites() {
    return [
      {
        source: ['/example/a', '/example/b', '/example/c'],
        destination: '/example'
      }
    ]
  }
}
```

**Comment by timneutkens** [comment]
This can be added to your next.config.js using one line of JS:

```js
module.exports = {
  async rewrites() {
    return [
      ... ['/example/a', '/example/b', '/example/c'].map(source => ({ source, destination: '/example' })
    ]
  }
}
```

**Reply by takeItIzzy** [reply]
thanks a lot

**Comment by aclimarin** [comment]
I tried this but i`m getting this error:

`source` is missing, `destination` is missing, `permanent` is not set to `true` or `false`, invalid fields: 0,1 for route [{"source":"/user","has":[{"type":"cookie","key":"cefauth.authorize","value":"false"}],"permanent":true,"destination":"/login"},{"source":"/dashboard","has":[{"type":"cookie","key":"cefauth.authorize","value":"false"}],"permanent":true,"destination":"/login"}]

`const protectedRoutes = ['/user', '/dashboard']`

`redirects: async () => {
    return [
      {
        source: '/',
        permanent: true,
        destination: '/login',
      },
      {
        source: '/login',
        has: [
          {
            type: 'cookie',
            key: 'cefauth.authorize',
            value: 'true',
          },
        ],
        permanent: true,
        destination: '/dashboard',
      },
      protectedRoutes.map(source => (
        {
          source,
          has: [
            {
              type: 'cookie',
              key: 'cefauth.authorize',
              value: 'false',
            },
          ],
          permanent: true,
          destination: '/login',
        }
      ))
    ]
  }`

**Reply by steroyle** [reply]
I had the same error. For me it was because I wasn't spreading the map results using `...`. Try adding it before your `...protectedRoutes.map()`.
